src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateBuildOptionsRule.cs

```diff
@@ -0,0 +1,405 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Construction;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.Cli;
+using System.Linq;
+using System.IO;
+using Newtonsoft.Json;
+using Microsoft.DotNet.ProjectModel.Files;
+using Microsoft.DotNet.Cli.Utils;
+using Newtonsoft.Json.Linq;
+
+using Project = Microsoft.DotNet.ProjectModel.Project;
+
+namespace Microsoft.DotNet.ProjectJsonMigration
+{
+    // TODO: Should All build options be protected by a configuration condition?
+    //       This will prevent the entire merge issue altogether and sidesteps the problem of having a duplicate include with different excludes...
+    public class MigrateBuildOptionsRule : IMigrationRule
+    {
+        private AddPropertyTransform<CommonCompilerOptions>[] EmitEntryPointTransforms
+            => new AddPropertyTransform<CommonCompilerOptions>[]
+            {
+                new AddPropertyTransform<CommonCompilerOptions>("OutputType", "Exe",
+                    compilerOptions => compilerOptions.EmitEntryPoint != null && compilerOptions.EmitEntryPoint.Value),
+                new AddPropertyTransform<CommonCompilerOptions>("TargetExt", ".dll",
```

Comment:
`TargetExt` is always set in the Sdk.props file.  See [https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.DotNet.Core.Build.Tasks/build/netstandard1.0/Microsoft.DotNet.Core.Sdk.props#L53-L54](https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.DotNet.Core.Build.Tasks/build/netstandard1.0/Microsoft.DotNet.Core.Sdk.props#L53-L54).